docs/DeveloperGuide.adoc

```diff
@@ -218,29 +228,431 @@ image::CommitActivityDiagram.png[]
 // end::undoredo[]
 
-// tag::dataencryption[]
-=== [Proposed] Data Encryption
+// tag::archive[]
+=== Archive
+==== Implementation
+===== Modelling Archive
+The archive feature supports the following actions:
+
+* Creating an archive
+* Added an expense to an archive
+* Reverting/"unarchiving" an archived expense
+* Deleting an archived expense
+* Displaying the list of expense of a particular archive
+* Listing all existing archive names
+
+
+These actions are facilitated by the `ArchiveWrapper` and `Archive` classes:
+
+*  `Archive` extends from `ExpenseList` in order to encapsulate an archive name and a list of expenses together as an archive.
+* `ArchiveWrapper` manages all existing archives and hashes each `Archive` object to its archive name.
+
+Similar to `Billboard`, which manages the list of current non-archived expenses,
+`ArchiveWrapper` is used in `ModelManager` and its respective operations are called to access and manipulate archive expenses when an archive command is entered. +
+Such operations include:
+
+* `ArchiveWrapper#AddArchive(Archive)` - Adds the given archive to the current map of archive objects.
+* `ArchiveWrapper#HasArchive(String)` - Checks if the specified archive by the given archive name exists.
+* `ArchiveWrapper#removeArchive(Archive)` - Deletes the given archive from the current map of archive objects. (Assumes given archive already exists)
+* `ArchiveWrapper#hasArchiveExpense(String, Expense)` - Checks if the specified archive by the given archive name has the given expense.
+* `ArchiveWrapper#addArchiveExpense(String, Expense)` - Adds the given expense into the specified archive by the given archive name. (Assumes given archive already exists)
+* `ArchiveWrapper#removeArchiveExpense(String, Expense)` - Deletes the given expense into the specified archive by the given archive name. (Assumes given archive and expense already exists)
+* `ArchiveWrapper#getArchiveNames()` - Returns a set of all existing archive names
+
+These operations are exposed in the `Model` interface respectively as:
+
+* `Model#addArchive(Archive)`
+* `Model#HasArchive(String)`
+* `Model#deleteArchive(Archive)`
+* `Model#hasArchiveExpense(String, Expense)`
+* `Model#addArchiveExpense(String, Expense)`
+* `Model#deleteArchiveExpense(String, Expense)`
+* `Model#getArchiveNames()`
+
+Given below is an example usage scenario of the add expense to archive function, showing how the command is parsed in `Logic` and how it interacts with `Model`: +
+
+Firstly, the user has the application running and has a non empty list of current expenses. The user can enter the `list` command to bring up
+this list. +
+
+Next, the user executes the command `archive add 3 arc/archiveName` to archive an expense. +
+The command is first parsed by `BillboardParser` to determine what kind of general command it is. `archive` indicates it is an archive command so the remaining input is parsed through `ArchiveCommandParser`. +
```

Comment:
This paragraph uses the same font and bolding for classes, arguments, commands, and quotations. This might be confusing to the reader as they might not know what each bolded word represents. Maybe separating out the font and standardizing would make it better.